src/Client/OAuth2Client.php

```diff
@@ -136,9 +145,17 @@ public function getOAuth2Provider()
     }
 
     /**
+     * @return bool
+     */
+    public function isStateless()
+    {
+        return $this->isStateless;
+    }
+
+    /**
      * @return \Symfony\Component\HttpFoundation\Request
      */
-    private function getCurrentRequest()
+    protected function getCurrentRequest()
```

Comment:
I actually disagree with this: we should only add extension points when people ask and give us a good reason - that'll help maintenance in the future. If we keep this private, the can still fetch the Request if they want to, but injecting the `request_stack` service into directly into some base class.